app/controllers/worksheets_controller.rb

```diff
@@ -0,0 +1,31 @@
+class WorksheetsController < ApplicationController
+
+  def show
+    @worksheet = Worksheet.find(params[:id])
+     # authorize @worksheet
+  end
+
+  def edit
+    @classroom = Classroom.find(params[:classroom_id])
+    @work_group = WorkGroup.find(params[:work_group])
+    @worksheet = Worksheet.find(params[:id])
+  end
+
+
+  def update
+    @classroom = Classroom.find(params[:classroom_id])
+    @work_group = WorkGroup.find(params[:work_group])
+    @worksheet = Worksheet.find(params[:id])
+    if @worksheet.update(worksheet_params)
+      # TODO redirect using JSON ?
+    else
+      render :edit
+  end
```

Comment:
you are missing and end here

Reply:
Thank you for spotting this. 
I have added the missing 'end'